commons-macros/src/main/scala/com/avsystem/commons/macros/serialization/GenCodecMacros.scala

```diff
@@ -231,7 +232,13 @@ class GenCodecMacros(ctx: blackbox.Context) extends CodecMacroCommons(ctx) with
       def readField(param: ApplyParam) = {
         val fieldName = nameBySym(param.sym)
         val defaultValueTree = param.defaultValue.orElse(q"fieldMissing($fieldName)")
-        q"${optName(param)}.getOrElse($defaultValueTree)"
+        if (!isScalaJs) q"${optName(param)}.getOrElse($defaultValueTree)"
+        else {
+          q"""
+            if (elements.containsKey(${optName(param).toString})) elements.get(${optName(param).toString}).asInstanceOf[${param.valueType}]
```

Comment:
Why not just use `getOrElse`? Are you trying to avoid by-name params? Also, it's weird that `optName` is used here - I would suppose it would be more natural to simply use `fieldName`.
Also, if you're using a map here then for consistency it should also be used in `OutOfOrderFields`